pkg/registry/apis/provisioning/safepath/walk.go

```diff
@@ -0,0 +1,31 @@
+package safepath
+
+import (
+	"context"
+	"path"
+	"strings"
+)
+
+type WalkFunc func(ctx context.Context, path string) error
+
+// Walk walks the given folder path and calls the given function for each folder.
+func Walk(ctx context.Context, p string, fn WalkFunc) error {
```

Comment:
I don't like the name here. It's entirely possible that at some point, we misunderstand this to be similar to `filepath.Walk`. Maybe `WalkFolderPath` or `TraverseFolderPath`?

Reply:
It's actually the same as `filepath.Walk` but without looking into the filesystem: 

If you see the `WalkFunc` there has a file in the signature: https://pkg.go.dev/path/filepath#WalkFunc. 

Maybe `WalkPath`? 

It's already in the `safepath` package that's why I was calling it only `Walk`. 

@Proximyst